src/monitor/node_active_protocol.c

```diff
@@ -965,7 +988,89 @@ stop_maintenance(PG_FUNCTION_ARGS)
 					  currentNode->nodePort,
 					  currentNode->pgsrSyncState,
 					  currentNode->reportedLSN,
+					  currentNode->candidatePriority,
+					  currentNode->replicationQuorum,
 					  message);
 
 	PG_RETURN_BOOL(true);
 }
+
+
+/*
+ * set_node_replication_settings sets node replication parameters.
+ * It could be called on both primary or standby nodes. The function
+ * returns true on success.
+ */
+Datum
+set_node_replication_settings(PG_FUNCTION_ARGS)
+{
+	int32 nodeid = PG_GETARG_INT32(0);
+	int candidatePriority = PG_GETARG_INT32(1);
+	bool replicationQuorum = PG_GETARG_BOOL(2);
+	AutoFailoverNode *currentNode = NULL;
+	bool nodeUpdated = false;
+
+
+	ereport(LOG, (errmsg("updating node replication properties for node %d, candidate_priority=%d, replication_quorum=%s",
+						  nodeid, candidatePriority, replicationQuorum?"true":"false")));
+
+	checkPgAutoFailoverVersion();
+
+	currentNode = GetAutoFailoverNodeWithId(nodeid);
+
+	if (currentNode == NULL)
+	{
+		ereport(ERROR, (errmsg("node %d is not registered", nodeid)));
+	}
+
+	LockFormation(currentNode->formationId, ShareLock);
+	LockNodeGroup(currentNode->formationId, currentNode->groupId, ExclusiveLock);
```

Comment:
We lock the node after having read the current values from the table. In between those two operations, a concurrent function call could have changed the values already. Is there a way that we don't depend on the current values that way in the implementation of this function?

Maybe on the monitor side too we should have a specific function per parameter. Or a way to not have to guess if there's a value that's been given but that we don't actually want to update...

Reply:
separated functions in both cli and monitor. My earlier intention was to set them in a single call, specify NULL for not setting the value. It did not work as I expected.  Now resort to separate calls for everything.